src/app/services/technical-record/technical-record.service.ts

```diff
@@ -117,6 +115,39 @@ export class TechnicalRecordService {
     return this.http.put<VehicleTechRecordModel>(url, body, { responseType: 'json' });
   }
 
+  clearReasonForCreation(vehicleTechRecord?: VehicleTechRecordModel): void {
+    this.editableVehicleTechRecord$
+      .pipe(
+        map(data => cloneDeep(data ?? vehicleTechRecord)),
+        take(1)
+      )
+      .subscribe(data => {
+        if (data) {
+          data.techRecord[0].reasonForCreation = '';
+          this.updateEditingTechRecord(data);
+        }
+      });
+  }
+
+  updateEditingTechRecord(record: TechRecordModel | VehicleTechRecordModel, resetVehicleAttributes = false): void {
```

Comment:
This is complex, and maybe not 100% clear what we are doing here or why you need it, is it worth a small documenting comment? Genuine question, happy either way

Reply:
I can add a documenting comment. Basically, I'm using the service to interact with the store, so you can pass either the whole vehicle object or just a tech record. This is to avoid making loads of changes to components e.g. passing the whole vehicle object to 10 components, and just keep using the techRecord with most components interact with. The service is then the one responsible for mapping the given `techRecord` to the vehicleRecord. There is an override that essentially resets the vehicle information to the information in state for the current vehicle, before being editing.